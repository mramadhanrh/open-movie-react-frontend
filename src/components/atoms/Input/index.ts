import styled from 'styled-components';

interface InputProps {
  width?: number;
}

const Input = styled.input<InputProps>`
  border: none;
  box-shadow: 0px 0px 0px 0px #f86f22;
  border-radius: 3px;
  background-color: ${({ theme }) => theme.color.mischka};
  width: ${({ width }) => width}px;
  padding: 10px;
  font-weight: 500;

  ::placeholder {
    font-weight: normal;
  }
`;

export default Input;
