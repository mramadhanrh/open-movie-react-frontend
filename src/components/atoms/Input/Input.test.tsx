import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Input from '.';
import defaultTheme from '../../../constants/theme';

describe('Input', () => {
  test('match snapshot', () => {
    expect(toJson(shallow(<Input theme={defaultTheme} />))).toMatchSnapshot();
  });

  test('controlled value match with input text', () => {
    const input = mount(
      <Input theme={defaultTheme} onChange={() => {}} value="Hello" />,
    );
    input.getDOMNode<HTMLInputElement>().value = 'Hi';
    input.simulate('change');
    expect(input.getDOMNode<HTMLInputElement>().value).toBe('Hello');
  });

  test('uncontrolled value match with input test', () => {
    const input = mount(
      <Input theme={defaultTheme} onChange={() => {}} defaultValue="Hello" />,
    );
    input.getDOMNode<HTMLInputElement>().value = 'Hi';
    input.simulate('change');
    expect(input.getDOMNode<HTMLInputElement>().value).toBe('Hi');
  });
});
