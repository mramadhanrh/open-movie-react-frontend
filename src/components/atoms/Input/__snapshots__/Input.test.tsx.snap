// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input match snapshot 1`] = `
<input
  className="sc-bdnxRM jaMaXq"
/>
`;
